### ISSUE TITLE
False positive ``superfluous-parens`` with walrus operator

### ISSUE THREAD

**Issue by otchita** [PyCQA/pylint#4346]
I've came across a `superfluous-parens` false positive when using the walrus operator:

### Steps to reproduce

Given a file `test.py`:
```python
"""Test example."""
if (var := 3) in [1, 2, 3]:
    print('Hello world !')
```

We obtain the following:
### Current behavior

Result of `pylint test.py`:
```
************* Module test
test.py:2:0: C0325: Unnecessary parens after 'if' keyword (superfluous-parens)
```

### Expected behavior

Normally, this should not raise any errors as the use of parentheses here is relevant. Because, otherwise the value of `var` will be the inclusion boolean instead of the value itself. 

### pylint --version output

Result of pylint --version output on my local macOS:

```
pylint 2.7.4
astroid 2.5.2
Python 3.8.8 (v3.8.8:024d8058b0, Feb 19 2021, 08:48:17) 
[Clang 6.0 (clang-600.0.57)]
```
